src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/RepresentativeService.java

```diff
@@ -38,11 +44,23 @@ public Representative getApplicantRepresentative(CaseData caseData) {
         var organisationId = caseData.getApplicant1OrganisationPolicy().getOrganisation().getOrganisationID();
         var representative = fromOrganisation(organisationService.findOrganisationById(organisationId)
                                                   .orElseThrow(RuntimeException::new));
+
+        var providedSolicitor1ServiceAddress = caseData.getApplicantSolicitor1ServiceAddress();
+
         return representative.toBuilder()
             .emailAddress(caseData.getApplicantSolicitor1UserDetails().getEmail())
+            .serviceAddress(fromProvidedAddress(representative.getServiceAddress(), providedSolicitor1ServiceAddress))
             .build();
     }
 
+    private Address fromProvidedAddress(
+        Address registeredServiceAddress, SolicitorServiceAddress providedServiceAddress) {
+        if (providedServiceAddress != null && providedServiceAddress.getRequired().equals(YesOrNo.YES)) {
+            return providedServiceAddress.getAddress();
+        }
+        return registeredServiceAddress;
+    }
```

Comment:
I would make a method for this on caseData, so that we can do caseData.getServiceAddress or something similar. Logic is then in one place rather than needed in multiple

Reply:
it's a generic method for three different Service addresses captured in claim creation, null check is temporary to make it backward compatible as well as it is going to be mandatory, however it can move in some common helper